compiler/rustc_mir_transform/src/known_panics_lint.rs

```diff
@@ -101,8 +101,14 @@ impl<'tcx> Value<'tcx> {
                 }
                 (PlaceElem::Index(idx), Value::Aggregate { fields, .. }) => {
                     let idx = prop.get_const(idx.into())?.immediate()?;
-                    let idx = prop.ecx.read_target_usize(idx).ok()?;
-                    fields.get(FieldIdx::from_u32(idx.try_into().ok()?)).unwrap_or(&Value::Uninit)
+                    let idx: u32 = prop.ecx.read_target_usize(idx).ok()?.try_into().ok()?;
+
+                    let max: u32 = FieldIdx::MAX.index().try_into().ok()?;
+                    if idx > max {
```

Comment:
Please implement this as a method on all index types instead. Like `try_from_u32` and then implement `from_u32` by using that

Reply:
Hmm... we could just always return an option, good point. Yea lets do that instead of my idea of having both a try_from and a from method

Reply:
I have a doubt here. In this match inside `project` method from `Value` type, why do we always cast the usize idx to a u32 just to call then `from_u32` when we have also a  `from_usize` method?

Reply:
probably just an oversight. using `from_usize` is better